x-pack/test/detection_engine_api_integration/security_and_spaces/roles_users_utils/index.ts

```diff
@@ -4,63 +4,62 @@
-import * as detectionsAdminRole from '../../../../plugins/security_solution/server/lib/detection_engine/scripts/roles_users/detections_admin/detections_role.json';
+import { assertUnreachable } from '../../../../plugins/security_solution/common/utility_types';
+import {
+  t1AnalystUser,
+  t2AnalystUser,
+  hunterUser,
+  ruleAuthorUser,
+  socManagerUser,
+  platformEngineerUser,
+  detectionsAdminUser,
+  readerUser,
+  t1AnalystRole,
+  t2AnalystRole,
+  hunterRole,
+  ruleAuthorRole,
+  socManagerRole,
+  platformEngineerRole,
+  detectionsAdminRole,
+  readerRole,
+} from '../../../../plugins/security_solution/server/lib/detection_engine/scripts/roles_users';
 
 import { ROLES } from '../../../../plugins/security_solution/common/test';
 import { FtrProviderContext } from '../../common/ftr_provider_context';
 
 export const createUserAndRole = async (
   securityService: ReturnType<FtrProviderContext['getService']>,
-  role: keyof typeof ROLES
-) => {
+  role: ROLES
+): Promise<void> => {
   switch (role) {
     case ROLES.detections_admin:
-      await postRoleAndUser(
+      return postRoleAndUser(
         ROLES.detections_admin,
         detectionsAdminRole,
         detectionsAdminUser,
         securityService
       );
-      break;
     case ROLES.t1_analyst:
-      await postRoleAndUser(ROLES.t1_analyst, t1AnalystRole, t1AnalystUser, securityService);
-      break;
+      return postRoleAndUser(ROLES.t1_analyst, t1AnalystRole, t1AnalystUser, securityService);
     case ROLES.t2_analyst:
-      await postRoleAndUser(ROLES.t2_analyst, t2AnalystRole, t2AnalystUser, securityService);
-      break;
+      return postRoleAndUser(ROLES.t2_analyst, t2AnalystRole, t2AnalystUser, securityService);
     case ROLES.hunter:
-      await postRoleAndUser(ROLES.hunter, hunterRole, hunterUser, securityService);
-      break;
+      return postRoleAndUser(ROLES.hunter, hunterRole, hunterUser, securityService);
     case ROLES.rule_author:
-      await postRoleAndUser(ROLES.rule_author, ruleAuthorRole, ruleAuthorUser, securityService);
-      break;
+      return postRoleAndUser(ROLES.rule_author, ruleAuthorRole, ruleAuthorUser, securityService);
     case ROLES.soc_manager:
-      await postRoleAndUser(ROLES.soc_manager, socManagerRole, socManagerUser, securityService);
-      break;
+      return postRoleAndUser(ROLES.soc_manager, socManagerRole, socManagerUser, securityService);
     case ROLES.platform_engineer:
-      await postRoleAndUser(
+      return postRoleAndUser(
         ROLES.platform_engineer,
         platformEngineerRole,
         platformEngineerUser,
         securityService
       );
-      break;
+    case ROLES.reader:
+      return postRoleAndUser(ROLES.reader, readerRole, readerUser, securityService);
     default:
-      break;
+      return assertUnreachable(role);
```

Comment:
NOTE: Using this and changing the enum above gives us exhaustive switches where when someone adds a new role type such as `ROLES.reader` (which was missing I noticed), they are forced to add it to this function as well.